src/main/java/com/example/application/services/ClientsService.java

```diff
@@ -0,0 +1,57 @@
+package com.example.application.services;
+
+import com.example.application.dao.ClientDAO;
+import com.example.application.models.ClientsEntity;
+import org.dozer.DozerBeanMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import com.example.application.dto.ClientDTO;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ClientsService {
+
+    private final DozerBeanMapper mapper;
+
+    private final ClientDAO clientDAO;
+
+    @Autowired
+    public ClientsService(ClientDAO clientDAO, DozerBeanMapper mapper) {
+        this.clientDAO = clientDAO;
+        this.mapper = mapper;
+    }
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public List<ClientDTO> getClientsList() {
+        return clientDAO.index().stream()
+                .map(entity -> mapper.map(entity, ClientDTO.class))
+                .collect(Collectors.toList());
+    }
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public ClientDTO getClient(int id) {
+        ClientsEntity client = clientDAO.show(id);
+        System.out.println(client.toString());
```

Comment:
It is not good practice to use `System.out.println`, try to use one of common java loggers